src/makielayout/blocks/polaraxis.jl

```diff
@@ -0,0 +1,537 @@
+function Makie.apply_transform(f::PolarAxisTransformation, r::Rect2{T}) where {T}
+    # TODO: once Proj4.jl is updated to PROJ 8.2, we can use
+    # proj_trans_bounds (https://proj.org/development/reference/functions.html#c.proj_trans_bounds)
+    N = 21
+    umin = vmin = T(Inf)
+    umax = vmax = T(-Inf)
+    xmin, ymin = minimum(r)
+    xmax, ymax = maximum(r)
+    # If ymax is 2π away from ymin, then the limits
+    # are a circle, meaning that we only need the max radius
+    # which is trivial to find.
+    # @show r
+    if abs(ymax - ymin) ≈ 2π
+        @assert xmin ≥ 0
+        rmax = xmax
+        # the diagonal of a square is sqrt(2) * side
+        # the radius of a circle inscribed within that square is side/2
+        mins = Point2f(-rmax)#Makie.apply_transform(f, Point2f(xmin, ymin))
+        maxs = Point2f(rmax*2)#Makie.apply_transform(f, Point2f(xmax - xmin, prevfloat(2f0π)))
+        @show(mins, maxs)
+        return Rect2f(mins,maxs)
+    end
+    for x in range(xmin, xmax; length = N)
+        for y in range(ymin, ymax; length = N)
+            u, v = Makie.apply_transform(f, Point(x, y))
+            umin = min(umin, u)
+            umax = max(umax, u)
+            vmin = min(vmin, v)
+            vmax = max(vmax, v)
+        end
+    end
+
+    return Rect(Vec2(umin, vmin), Vec2(umax-umin, vmax-vmin))
+end
+
+
+# Define its inverse (for interactivity)
+Makie.inverse_transform(trans::PolarAxisTransformation) = Makie.PointTrans{2}() do point
+    Point2f(hypot(point[1], point[2]), -trans.direction * (atan(point[2], point[1]) - trans.θ_0))
+end
+
+# End transform code
+
+# Some useful code to transform from data (transformed) space to pixelspace
+
+function project_to_pixelspace(scene, point::VecTypes{N, T}) where {N, T}
+    @assert N ≤ 3
+    return to_ndim(
+        typeof(point),
+        Makie.project(
+            # obtain the camera of the Scene which will project to its screenspace
+            camera(scene),
+            # go from dataspace (transformation applied to inputspace) to pixelspace
+            :data, :pixel,
+            # apply the transform to go from inputspace to dataspace
+            Makie.apply_transform(
+                scene.transformation.transform_func[],
+                point
+            )
+        ),
+        0.0
+    )
+end
+
+function project_to_pixelspace(scene, points::AbstractVector{Point{N, T}}) where {N, T}
+    to_ndim.(
+        Ref(eltype(points)),
+        Makie.project.(
+            # obtain the camera of the Scene which will project to its screenspace
+            Ref(Makie.camera(scene)),
+            # go from dataspace (transformation applied to inputspace) to pixelspace
+            Ref(:data), Ref(:pixel),
+            # apply the transform to go from inputspace to dataspace
+            Makie.apply_transform(
+                scene.transformation.transform_func[],
+                points
+            )
+        ),
+        Ref(0.0)
+    )
+end
```

Comment:
I'm not sure if it makes much of a difference but I think `(val,)` is preferable over `Ref(val)` in these cases because it doesn't allocate.

Also `project(scene, apply_transform(f, point))` (or the lower level variants) should be used here and above. The variants that specify input and output space have some extra branching and matrix multiplications in them

Reply:
Is there a difference between that and what's done here, aside from code cleanliness?  Basically, I didn't want to call `camera` and `apply_transform` in an inner loop so did this instead.

Also, AFAIK since the transform_func is an Observable{Any}, having apply_transform in an inner loop takes a lot of time compared to having it outside.